### PR TITLE
Reexecute all last failed tests regardless of source change status (tlf)

### PR DESCRIPTION
**Pull request by tarpas**
--testmon --tlf forces reexecution of all failures even if the underlying source code didn't change. This serves if you wan't different error output or if you corrupted .testmondata through unaccesible external service or wrong virtualenv.

### CODE REVIEW

**Comment by tarpas** [comment]
Documentation missing.

**Comment by blueyed** [comment]
Thanks!

In the commit message: s/wan't/want/

File: test/test_testmon.py [hunk 1]
```diff
@@ -192,6 +192,25 @@ def test_add():
...
+                1/0
+        """, )
+        testdir.inline_run("--testmon", "-v")
+        sys.modules.pop('test_a', None)
+
+        result = testdir.runpytest("--testmon", "-v")
+        result.stdout.fnmatch_lines([
+            "*1 failed, 1 deselected*",
+        ])
+        tf.setmtime(1424880936)
```
Comment (blueyed) [inline test/test_testmon.py]: Could benefit from a comment that this is in the past (`datetime.datetime(2015, 2, 25, 17, 15, 36)`).

File: testmon/pytest_testmon.py [hunk 2]
```diff
@@ -42,6 +42,13 @@ def pytest_addoption(parser):
     )
 
     group.addoption(
+        '--tlf',
+        action='store_true',
+        dest='tlf',
+        help="Re-execute last failures regardless of source change status",
```
Comment (blueyed) [inline testmon/pytest_testmon.py]: You might want to use single quotes throughout the diff.. :)

File: test/test_testmon.py [hunk 3]
```diff
@@ -192,6 +192,25 @@ def test_add():
         assert tuple(res) == (1, 0, 0), res
         sys.modules.pop('test_a', None)
 
+    def test_tlf(self, testdir):
+        tf = testdir.makepyfile(test_a="""
+            def test_add():
+                1/0
+        """, )
+        testdir.inline_run("--testmon", "-v")
+        sys.modules.pop('test_a', None)
```
Comment (blueyed) [inline test/test_testmon.py]: Why is that necessary?
Could benefit from a comment.

File: test/test_testmon.py [hunk 4]
```diff
@@ -192,6 +192,25 @@ def test_add():
...
+
+        result = testdir.runpytest("--testmon", "-v")
+        result.stdout.fnmatch_lines([
+            "*1 failed, 1 deselected*",
+        ])
+        tf.setmtime(1424880936)
+        result = testdir.runpytest("--testmon", "-v", "--tlf")
+        result.stdout.fnmatch_lines([
+            "*1 failed in*",
+        ])
```
Comment (blueyed) [inline test/test_testmon.py]: To simulate the behavior of an external service (e.g. DB) failing, you could have a test that fails if a file does not exist, and then create it later - where it then should pass with `--tlf`.

File: test/test_testmon.py (hunk 1 above)
Comment (tarpas) [inline test/test_testmon.py]: I'll leave it like that, as I don't think a comment would make it much better. I created an issue: https://github.com/tarpas/pytest-testmon/issues/72

File: test/test_testmon.py (hunk 4 above)
Comment (tarpas) [inline test/test_testmon.py]: That's the main use case, but I think this is a better minimal test case. We execute identical, failing, test 3 times. 1. test - > creating fixture, 2. test -> we verify that it was deselected because it didnt' change, 3. test -> we verify it was executed (not deselected) because of the --tlf switch, even when not changed.

Comment (tarpas) [inline test/test_testmon.py]: If you wanted a more involved test case with external factors I would recommend somthing like:
  ```python
    os.environ['TLF']='ON'
    tf = testdir.makepyfile(test_a="""
            import os
            def test_add():
                if os.environ.get('TLF', None):
                    raise Exception('TLFON')
        """, )
    testdir.inline_run("--testmon", "-v")

```
